src/app/show-one-product/show-one-product.component.ts

```diff
@@ -1,32 +1,36 @@
 import { Component, OnInit } from '@angular/core';
 import {ActivatedRoute, Router} from '@angular/router';
-import {Product} from '../model/product.model';
-import {ProductService} from '../../services/product.services';
+import {Product} from '../store/state/product.state';
+import {AppState} from "../store/state/app.state";
+import {select, Store} from "@ngrx/store";
+import {selectSelectedProduct} from "../store/selectors/product.selectors";
+import {DeleteProduct, GetProduct} from "../store/actions/product.actions";
 
 
 
 @Component({
   selector: 'app-show-one-product',
-  templateUrl: './show-one-product.component.html',
-  styleUrls: ['./show-one-product.component.css']
+  templateUrl: './show-one-product.component.html'
 })
 export class ShowOneProductComponent implements OnInit {
 
   productId = '';
-  product: Product;
+  product$ = this.store.pipe(select(selectSelectedProduct));
```

Comment:
It's usually a good practice to initialize this later in the flow, for instance in the ngOnInit function.